externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/PlanOnlyStatement.scala

```diff
@@ -38,23 +38,25 @@ class PlanOnlyStatement(
   private val operationLog: OperationLog = OperationLog.createOperationLog(session, getHandle)
   override def getOperationLog: Option[OperationLog] = Option(operationLog)
 
-  override protected def resultSchema: StructType =
-    if (result == null) {
+  override protected def resultSchema: StructType = Option(result) match {
+    case Some(res) if res.isEmpty =>
+      new StructType().add("Result", "string")
+    case Some(res) =>
+      res.schema
+    case _ =>
       new StructType().add("plan", "string")
-    } else {
-      result.schema
-    }
+  }
 
-  private def isSetOrReset(plan: LogicalPlan): Boolean = {
+  private def shouldDirectRun(plan: LogicalPlan): Boolean = {
     val className = plan.getClass.getSimpleName
-    className == "SetCommand" || className == "ResetCommand"
+    className == "SetCommand" || className == "ResetCommand" || className == "UseStatement"
```

Comment:
I did't find `UseStatement` at spark's master branch

Reply:
it has migrated to `SetNamespaceCommand`, see https://github.com/apache/spark/pull/34127

Reply:
In Spark 3.0, `UseStatement` is replaced with `SetNamespaceCommand`, which changes in [https://github.com/apache/spark/pull/34127](https://github.com/apache/spark/pull/34127).

Reply:
> h migrate `UseStatement` to v2 command framework and change it to `SetNamespaceCommand`, But Spark 3.1 and 3.2 still have `UseStatement`. Do our changes here need to be forward compatible like Spark 3.3?

IMO, for compatibility, you could check whether the Spark version has the `UseStatement`. If no, use the `SetNamespaceCommand`.

Reply:
we can match them all in this PR, we have a daily flow to check compatibility for spark master